### PR TITLE
Add integration tests for diff logging

### PR DESCRIPTION
**Pull request by dhaiducek**
ref: https://issues.redhat.com/browse/ACM-9072 (Dev issue)
ref: https://issues.redhat.com/browse/ACM-9074 (QE issue)

### CODE REVIEW

File: test/resources/diff_generation/diff_generation.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  name: placement-policy-diff-gen-configmap
+  kind: PlacementRule
+  apiGroup: apps.open-cluster-management.io
+subjects:
+- name: policy-diff-gen-configmap
+  kind: Policy
+  apiGroup: policy.open-cluster-management.io
+---
+apiVersion: apps.open-cluster-management.io/v1
+kind: PlacementRule
```
Comment (dhaiducek) [inline test/resources/diff_generation/diff_generation.yaml:47]: Should this be a `Placement`? Probably. But I felt like implementing that was outside the scope of this PR.

**Comment by dhaiducek** [comment]
It passed on the very first run??? What are the chances! 🥳

File: test/resources/diff_generation/diff_generation.yaml (hunk 1 above)
Comment (gparvin) [reply test/resources/diff_generation/diff_generation.yaml:47]: Yes, maybe we need a between release work item to make tests with Placement easy to add.

Comment (dhaiducek) [reply test/resources/diff_generation/diff_generation.yaml:47]: Agreed. It should be straightforward in the end, especially if we wipe placement from the policy-collection and then have a common function to deploy placement for everything, but it'll be a bit of work to clean it all up!

Comment (dhaiducek) [reply test/resources/diff_generation/diff_generation.yaml:47]: That said, maybe it'd all get wrapped into this issue? https://issues.redhat.com/browse/ACM-9081

**Review by gparvin — APPROVED** [review]

**Pull request merged**